pkg/netconfig/netutils.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package netconfig
+
+import (
+	"bytes"
+	"os/exec"
+)
+
+type NetUtilsCommandExecutor interface {
+	ExecuteIPTablesCommand(ipVersion string, args ...string) error
+	ExecuteIPRouteCommand(ipVersion string, args ...string) (*bytes.Buffer, error)
+	ExecuteIPSetCommand(args ...string) error
+}
+
+type OSNetUtilsCommandExecutor struct{}
+
+func (r *OSNetUtilsCommandExecutor) ExecuteIPRouteCommand(ipVersion string, args ...string) (*bytes.Buffer, error) {
+	args = append([]string{"-" + ipVersion}, args...)
+	cmd := exec.Command("ip", args...)
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	return &out, err
+}
+
+func (r *OSNetUtilsCommandExecutor) ExecuteIPTablesCommand(ipVersion string, args ...string) error {
```

Comment:
Do you have considered handling iptables locking errors?

Reply:
`iptables` is called with the `-w`, meaning it waits until the lock is released.
Unlike my first assumption it waits forever and there is no default of 5s.
Should we add a maximum wait time?
However, if some process keeps the log, we will see some more problems on that node.